v1/bundle/bundle.go

```diff
@@ -715,8 +715,10 @@ func (r *Reader) Read() (Bundle, error) {
 	popts.RegoVersion = bundle.RegoVersion(popts.EffectiveRegoVersion())
 	for _, mf := range modules {
 		modulePopts := popts
-		if modulePopts.RegoVersion, err = bundle.RegoVersionForFile(mf.RelativePath, popts.EffectiveRegoVersion()); err != nil {
+		if regoVersion, err := bundle.RegoVersionForFile(mf.RelativePath, popts.EffectiveRegoVersion()); err != nil {
 			return bundle, err
+		} else if regoVersion != ast.RegoUndefined {
+			modulePopts.RegoVersion = regoVersion
```

Comment:
[comment]: It is not obvious to me exactly when this condition could occur, even after diving into the sources for `bundle.RegoVersionForFile`. As best I can tell though, getting *both* an error and a `regoVersion` that isn't the default `ast.RegoUndefined` value *should* be pretty difficult.

If there's an obvious edge-case I'm missing, could we add a quick comment to explain it here? :sweat_smile:

Reply:
This is merely a safeguard against the world changing around us. You're correct that `ast.RegoUndefined` isn't an expected value here, but I think this is buried deeply enough that a future code change that alters this expectation could cause an issue here that is hard to detect. Also, if we get a `ast.RegoUndefined` returned here, we know that the correct course of action is to let the configured parser-options take precedence.

I'll add a comment 👍